### DISCUSSION TITLE
Delay Message Delivery

### DISCUSSION THREAD

**Discussion by satyendrakumarsingh** [emqx/emqx#9646] [Q&A]
I am testing my EMQX(5.0.11) setup and found delay in message acknowledge for random message.

1. Broker trace log show message published but receiver receive message with 5-10 / 30 seconds later and then PUBACK received for same.
2. This is not a regular behavior, I tried to replicate on different client and able to re-produce in 1/2 days. 

I just want to know how can I troubleshoot this latency as there is no such way to look other than trace log.

**Note**: There is no delay publish configuration in EMQX.

**Comment by zhangxin17610860215** [comment]
I have also encountered the same problem. If there is a solution, please let me know. Thank you very much!!!

**Comment by lafirest** [comment]
@satyendrakumarsingh 
Thanks for your feedback, would you please show me a possible way to reproduce this problem or paste your trace log file here, this could help us to dig up this problem.
